src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -24,14 +27,16 @@ public SafeFileHandle(IntPtr preexistingHandle, bool ownsHandle) : this(ownsHand
             SetHandle(preexistingHandle);
         }
 
-        internal bool? IsAsync { get; set; }
+        public bool IsAsync { get; private set; }
+
+        internal bool CanSeek => !IsClosed && (_canSeek ??= Interop.Sys.LSeek(this, 0, Interop.Sys.SeekWhence.SEEK_CUR) >= 0);
```

Comment:
Note that Nullable is not thread safe. Do you expect this property to be called from multiple threads?